### ISSUE TITLE
Ability to define different filesystem storage rules for different media use terms on ingest?

### ISSUE THREAD

**Issue by dara2** [mjordan/islandora_workbench#489]
In order for our thumbnail images to not appear broken for anonymous users on first page load, we store them in Drupal's public filesystem. We do this via a rule in the action that creates the thumbnail derivatives. However, we realized if we ingest objects and our Workbench CSV has a column for source media/files and a column for thumbnail images, those thumbnails are being put into Fedora. I saw that you can define the drupal_filesystem in the config file, but that will apply to ALL files being ingested (and is deprecated). We'd like to take advantage of Workbench's ability to add multiple media per object on ingest. 

It would be great if in our config file we could define the filesystem where files should go based on their Islandora Media Use term (e.g., Thumbnail Image). Otherwise, we will still need to ingest thumbnail images separately.

**Comment by mjordan** [comment]
@dara2 the `drupal_filesystem` doesn't do anything if you're using Drupal 9.2 or later. With that version, Workbench started using Drupal's REST interface for media and moved away from Islandora's, which proved problematic.

To my knowledge, Drupal's REST interface for file fields does not allow you to override the filesystem that is configured in the file field's configuration, but I will investigate to confirm that.

In the meantime, can you confirm that the filesystem configured for the thumbnail media is set to Drupal's public filesystem? If that's the case (sounds like it is), then that configuration setting is not being respected. But, Workbench is not in any way favoring the fedora filesystem, so we'll need to figure out why your thumbnails are being stored there.

**Comment by dara2** [comment]
Oh, interesting! Yes, looking at the Image field settings, the upload destination is Flysystem.

**Comment by dmer** [comment]
Image media-type is set for Flysystem - and that isn't wrong - that's where we want them stored UNLESS they are media-use-type Thumbnail so there's not an easy config fix for this that I can see.

**Comment by mjordan** [comment]
I've written a simple script to create a file via Drupal's REST interface, specifying the Drupal "public://" filesystem and destination path to the file to be in the Drupal public file space. In this example, it's an 'image' media whose 'field_media_image' field is configured to use the 'Flysystem: fedora' filesystem. I'm sorry to say that the script created the image with no errors, but the filesystem it created it on was 'Flysystem: fedora', not public:// (to be specific, its URL is http://localhost:8000/_flysystem/fedora/2022-11/austin.jpg). I'm pasting in the script here in case someone else wants to take a crack at it.

TLDR is that I do not know if it is possible for a REST client to override the filesystem configured for a media file field. This test, if valid, suggests that it is not possible.

```yaml
import requests

filename = 'austin.jpg'
media_type = 'image'
media_file_field = 'field_media_image'

binary_file_path = '/home/mark/hacking/islandora_workbench/austin.jpg'
username = 'admin'
password = 'islandora'

binary_data = open(binary_file_path, 'rb').read()

url = 'http://localhost:8000/file/upload/media/' + media_type + '/' + media_file_field + '?_format=json'

headers = {
    'Content-Type': 'application/octet-stream',
    'Content-Disposition': 'file; filename="' + filename + '"'
}

json = {'uri': [{'value': 'public://' + filename, 'url': '/sites/default/files/' + filename}]}

response = requests.post(url, auth=(username, password), headers=headers, json=json, data=binary_data)

print(f"Response: {response.status_code}")
```

**Comment by mjordan** [comment]
@dmer do you want to open a use case issue or would you like me to?

**Comment by dmer** [comment]
Thanks for that helpful test info Mark! We're going to discuss this issue internally then we'll be happy to open or contribute to an issue. If we open one, I'll put a link on this thread.

**Comment by dara2** [comment]
Yes, thank you @mjordan ! I will open an issue if needed.

**Comment by mjordan** [comment]
@dara2 @dmer anything to report back on this issue?

**Comment by dara2** [comment]
Hi @mjordan - we're working on fixing this internally (an action to run to fix files that are already in the "wrong" place, and some way to prevent them from going to Fedora in the first place). We've made progress on the first part, but not the second part yet. If there's not a way to address this via REST, we can probably close this issue.

**Comment by mjordan** [comment]
OK, thank you.
